unified-codes/apps/data-service/src/app/utils.ts

```diff
@@ -0,0 +1,25 @@
+import { IPaginatedResults, IPaginationParameters } from '@unified-codes/data';
+
+export function getPaginatedResults<T>(params: IPaginationParameters<T>): IPaginatedResults<T> {
+  const { results } = params;
+  const totalResults = results.length;
+  const entities = paginateResults(params);
+  const entityCount = entities.length;
+
+  return {
+    entities,
+    hasMore: totalResults > entityCount,
+    totalResults,
+  };
+}
+
+function paginateResults<T>(params: IPaginationParameters<T>): Array<T> {
+  const { first, offset, results } = params;
+
+  if (first < 1) return [];
+  if (!offset) return results.slice(0, first);
+  // don't let us overflow
+  if (offset === results.length - 1) return [];
+
+  return results.slice(offset + 1, Math.min(results.length, offset + 1 + first));
```

Comment:
Hmm offset is behaving funny for me 🙀  - I think it may be off by one? 
E.g. 
An offset of 1 is returning results starting at the 3rd item in the full set.
Also if offset is specified without a first value currently seeing an empty entities array (I think because first is undefined - maybe needs to default to 0?)

Reply:
nicely spotted. I missed the off-by-one! 
created when I refactored the cursors. thanks